test/common/grpc/grpc_client_integration_test.cc

```diff
@@ -29,6 +29,44 @@ TEST_P(GrpcClientIntegrationTest, BasicStream) {
   dispatcher_helper_.runDispatcher();
 }
 
+// Validate that a simple request-reply stream works with bytes metering in Envoy gRPC.
+TEST_P(GrpcClientIntegrationTest, BasicStreamWithBytesMeter) {
+  // Currently, only Envoy gRPC's bytes metering is based on the bytes meter in Envoy.
+  // Therefore, skip the test for google gRPC.
+  SKIP_IF_GRPC_CLIENT(ClientType::GoogleGrpc);
+  // The check in this test is based on HTTP2 codec logic (i.e., including H2_FRAME_HEADER_SIZE).
+  // Skip this test if default protocol of this integration test is no longer HTTP2.
+  if (fake_upstream_config_.upstream_protocol_ != Http::CodecType::HTTP2)
+    return;
+  initialize();
+  auto stream = createStream(empty_metadata_);
+
+  // Create the send request.
+  helloworld::HelloRequest request_msg;
+  request_msg.set_name(HELLO_REQUEST);
+
+  RequestArgs request_args;
+  request_args.request = &request_msg;
+  stream->sendRequest(request_args);
+  stream->sendServerInitialMetadata(empty_metadata_);
+
+  auto send_buf = Common::serializeMessage(request_msg);
+  Common::prependGrpcFrameHeader(*send_buf);
+
+  auto upstream_meter = stream->grpc_stream_->streamInfo().getUpstreamBytesMeter();
```

Comment:
So this support basically existed already and we're just avoiding doing internal book keeping now?

Reply:
Yes, exactly. I think it is expected since underneath the Envoy gRPC is http stream impl.

Basically, our issue is ext_proc was looking at wrong byte info e.g., [bytesReceived()](https://github.com/envoyproxy/envoy/blob/69595b4534f86eea4d87c742171ccbf806f209d2/source/common/stream_info/stream_info_impl.h#L186). I will fix this issue along with other changes and integration test in #28136

The PR here is mostly validating the codec-level bytes meter in the grpc test. I reverted the fixes for internal bytes booking e.g., [bytes_sent](https://github.com/envoyproxy/envoy/blob/69595b4534f86eea4d87c742171ccbf806f209d2/source/common/stream_info/stream_info_impl.h#L447) etc. from this PR because 1) we won't use it 2) it is outside scope of this PR. It is probably still good to fix them. I can do it in a separate PR

Reply:
Just FYI, #28136 is ready for review. I added a test filter(`logging_test_filter`) as the consumer for the bytes info that is provided by ext_proc.